tests/python/unittest/test_gluon_estimator.py

```diff
@@ -0,0 +1,238 @@
+                              metrics=acc,
+                              context=ctx)
+    est.fit(train_data=train_data,
+            epochs=num_epochs,
+            batch_size=batch_size)
+
+    # different initializer for net and estimator
+    net = get_model()
+    net.initialize(mx.init.Xavier(), ctx=ctx)
+    trainer = gluon.Trainer(net.collect_params(), 'sgd', {'learning_rate': 0.001})
+    # catch reinit warning
+    with warnings.catch_warnings(record=True) as w:
+        est = estimator.Estimator(net=net,
+                                  loss=loss,
+                                  metrics=acc,
+                                  initializer=mx.init.MSRAPrelu(),
+                                  trainers=trainer,
+                                  context=ctx)
+        assert 'Network already initialized' in str(w[-1].message)
+    est.fit(train_data=train_data,
+            epochs=num_epochs,
+            batch_size=batch_size)
+
+def test_trainer():
+    ''' test with no trainer and invalid trainer '''
+    net = get_model()
+    num_epochs = 1
+    batch_size = 4
+    ctx = mx.cpu()
+    in_data = mx.nd.random.uniform(shape=(10, 3))
+    out_data = mx.nd.random.uniform(shape=(10, 4))
+    dataset = gluon.data.dataset.ArrayDataset(in_data, out_data)
+    train_data = gluon.data.DataLoader(dataset, batch_size=batch_size)
+    loss = gluon.loss.L2Loss()
+    acc = mx.metric.Accuracy()
+    net.initialize(ctx=ctx)
+    # input no trainer
+    with warnings.catch_warnings(record=True) as w:
+        est = estimator.Estimator(net=net,
+                                  loss=loss,
+                                  metrics=acc,
+                                  context=ctx)
+        assert 'No trainer specified' in str(w[-1].message)
+    est.fit(train_data=train_data,
+            epochs=num_epochs,
+            batch_size=batch_size)
+
+    # input invalid trainer
+    trainer = 'sgd'
+    with assert_raises(ValueError):
+        est = estimator.Estimator(net=net,
+                                  loss=loss,
+                                  metrics=acc,
+                                  trainers=trainer,
+                                  context=ctx)
+
+def test_metric():
+    ''' test with no metric, list of metrics, invalid metric '''
+    net = get_model()
+    num_epochs = 1
+    batch_size = 4
+    ctx = mx.cpu()
+    in_data = mx.nd.random.uniform(shape=(10, 3))
+    out_data = mx.nd.random.uniform(shape=(10, 4))
+    dataset = gluon.data.dataset.ArrayDataset(in_data, out_data)
+    train_data = gluon.data.DataLoader(dataset, batch_size=batch_size)
+    loss = gluon.loss.L2Loss()
+    net.initialize(ctx=ctx)
+    trainer = gluon.Trainer(net.collect_params(), 'sgd', {'learning_rate': 0.001})
+    # input no metric
+    est = estimator.Estimator(net=net,
+                              loss=loss,
+                              trainers=trainer,
+                              context=ctx)
+    est.fit(train_data=train_data,
```

Comment:
What are we asserting against here ?

Reply:
this checks if the estimator works with no metric specified, doesn't throw any error/warning when its successful

Reply:
So there's a particular happy path that you're trying to test here ?